### PR TITLE
DOC: Add reference to dtype parameter in NDArray

### LINKED ISSUE #23722: DOC: Document first argument to `numpy.typing.NDArray`
### Issue with current documentation:

`numpy.typing.NDArray` takes two arguments when used as a type hint: https://numpy.org/doc/1.21/reference/typing.html#numpy.typing.NDArray

```python
numpy.typing.NDArray = numpy.ndarray[typing.Any, numpy.dtype[+ScalarType]]
```

The second argument seems to be the `dtype` (but isn't documented explicitly), and I'm not sure what the first argument is. 

### Idea or request for content:

The type hint arguments should be clearly documented on both `numpy.ndarray` and `numpy.typing.NDArray`.

### PR DESCRIPTION
**Pull request by undermyumbrella1**
Fixes #23722

### CODE REVIEW

File: numpy/_typing/_add_docstring.py
Comment (BvB93) [inline numpy/_typing/_add_docstring.py:1]: I'd reword things a little bit different (as the grammar is a bit wonky now) but besides that it looks fine.
Maybe something like:
```
A  `np.ndarray[Any, np.dtype[+ScalarType]] <numpy.ndarray>` type alias :term:`generic <generic type>` w.r.t. its `dtype.type <numpy.dtype.type>`.
```

Comment (ellaella12) [reply numpy/_typing/_add_docstring.py:1]: Resolved, thanks!

**Comment by BvB93** [comment]
Thanks @ellaella12.
